packages/x-scheduler/src/primitives/utils/adapter/useAdapter.ts

```diff
@@ -0,0 +1,7 @@
+import { getAdapter } from './getAdapter';
+
+export function useAdapter() {
+  // Temporary hook that mimics the behavior of a future `useAdapter` hook,
+  // to ease migration once the provider setup is ready.
+  return getAdapter();
```

Comment:
Just to make sure we don't have weird memoization issue

```suggestion
  return adapter;
```